torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -161,6 +161,38 @@ static inline at::optional<int64_t> parse_as_integer(const std::string& s) {
   return (*str_end == 0) ? at::optional<int64_t>(ans) : at::nullopt;
 }
 
+/*
+Parse default value of IntList declared at native_functions.yaml
+
+There are two kinds of default values:
+1. IntList[2] x=1 (where size=2, value={1,1}
+2. IntList x={1,2,3} (where size=3, value={1,2,3}, note that there cannot be space after comma since native_parse.py uses ', ' to split args)
+*/
+static inline std::vector<int64_t> parse_intlist_args(const std::string& s, int64_t size) {
+  size_t n = s.size();
+
+  if (s.empty()) return std::vector<int64_t>();
+
+  // case 1. s is an int (e.g., s=2)
+  if (s[0] != '{') {
+    return std::vector<int64_t>(size, std::stol(s));
+  }
+
+  // case 2. s is a list of dims (e.g., s={1,2})
+
+  // since already checked left brace '{' above, here only checks right brace '}'
+  AT_CHECK(s[n - 1] == '}', "Default value of IntList is missing right brace '}', found ", s[n - 1]);
+
+  auto args = std::vector<int64_t>();
+  const char* del = ",{}";
+  char* s_p = strdup(s.c_str());
```

Comment:
you are leaking this string.

Reply:
Great catch! Thanks for pointing that out!